lib/teselagen/api/tests/test_test_client.py

```diff
@@ -127,24 +132,29 @@ def assay_id(self) -> int:
     @pytest.fixture
     def experiment(
         self,
-        logged_client: TESTClient,
+        logged_client: TeselaGenClient,
         select_laboratory,
     ) -> Dict[str, Any]:
-        client = logged_client
+        client = logged_client.test
         experiment_name: str = "Python Test Client Experiment"
         experiment: Dict[str, Any] = client.create_experiment(
             experiment_name=experiment_name)
-        return experiment
+        
+        yield experiment
+
+        # Tear down
+        experiment_id: str = experiment['id']
+        client.delete_experiment(experiment_id=experiment_id)
```

Comment:
Nice! so now each experiment is created, yield it and deleted after the test has been executed 🥇 

---

I have not tried it yet, but - If you want to try it - I think that maybe the following could work to get the name of the test being executed and that it's using this fixture, which could for example be added to the name of the experiment, optionally with a timestamp.

```python3
from datetime import datetime

    @pytest.fixture
    def experiment(
        self,
        request,
        logged_client: TeselaGenClient,
        select_laboratory,
    ) -> Dict[str, Any]:
        # Uses the PyTest's `request` special fixture to get access to the test name
        
        #     request:
        #         https://docs.pytest.org/en/latest/reference/reference.html?highlight=request#request
        
        #     node:
        #         https://docs.pytest.org/en/latest/reference/reference.html?highlight=request#node
        
        test_name: str = request.node.name
        
        client = logged_client.test
        experiment_name: str = f"Python Test Client Experiment - Test: {test_name} - Run at: {str(datetime.now())} "
        
        experiment: Dict[str, Any] = client.create_experiment(
            experiment_name=experiment_name)

        yield experiment

        # Tear down
        experiment_id: str = experiment['id']
        client.delete_experiment(experiment_id=experiment_id)
```

I do not know if there are any restrictions on the platform regarding the names of the experiments, but perhaps it can be practical to include some metadata in relation to which test created which experiment and when exactly it occurred.